src/types/zodTypes.ts

```diff
@@ -0,0 +1,172 @@
+import { z } from 'zod';
+import { isDefinedAndNotNullAndNotEmpty } from '@rudderstack/integrations-lib';
+import { isHttpStatusSuccess } from '../v0/util';
+
+export const ProxyMetadataSchema = z.object({
+  jobId: z.number(),
+  attemptNum: z.number(),
+  userId: z.string(),
+  sourceId: z.string(),
+  destinationId: z.string(),
+  workspaceId: z.string(),
+  secret: z.record(z.unknown()),
+  destInfo: z.object({}).optional(),
+  omitempty: z.record(z.unknown()).optional(),
+  dontBatch: z.boolean(),
+});
+
+export const ProxyV0RequestSchema = z.object({
+  version: z.string(),
+  type: z.string(),
+  method: z.string(),
+  endpoint: z.string(),
+  userId: z.string(),
+  headers: z.record(z.unknown()).optional(),
+  params: z.record(z.unknown()).optional(),
+  body: z
+    .object({
+      JSON: z.record(z.unknown()).optional(),
+      JSON_ARRAY: z.record(z.unknown()).optional(),
+      XML: z.record(z.unknown()).optional(),
+      FORM: z.record(z.unknown()).optional(),
+    })
+    .optional(),
+  files: z.record(z.unknown()).optional(),
+  metadata: ProxyMetadataSchema,
```

Comment:
config is added to V0 request schema as well recently.

Reply:
Was it I thought it was only added to V1? Can you share link to server code once for my ref

Reply:
https://github.com/rudderlabs/rudder-server/pull/4352

Still in PR. More recent change done for trade desk destination.

Reply:
This has been addressed [here](https://github.com/rudderlabs/rudder-transformer/pull/3039/commits/99f5cb27328cb3595ffbd7f3a6a9f26ba112ca17)